prow/tide/status.go

```diff
@@ -338,7 +339,7 @@ func (sc *statusController) setStatuses(all []PullRequest, pool map[string]PullR
 			return baseSHA, nil
 		}
 
-		cr, err := sc.config().GetTideContextPolicy(sc.gc, org, repo, branch, baseSHAGetter, headSHA)
+		cr, err := getContextCheckerWithRequiredContexts(sc.config(), sc.gc, org, repo, branch, baseSHAGetter, headSHA, requiredContexts[prKey(pr)])
```

Comment:
This function doesn't seem necessary. Why not just do:
```golang
cr, err := sc.config().GetTideCP(sc.gc, org, ..)
cr.RequiredContexts = requiredContexts[prKey(pr)]
```
This function is only used once, and seems pretty trivial

Reply:
Yeah that's what I did initially, the reason I factored it the current way is that I do not know if overriding the `RequiredContexts` will break other parts of the `*TideContextPolicy` and wanted to make sure that ppl will only use the parts of it that implement the `ContextChecker`.
Not a very strong opinion though so I can change it back if you want me to